python/lsst/ts/mtdome/mtdome_csc.py

```diff
@@ -321,13 +324,21 @@ async def write_then_read_reply(
         st = encoding_tools.encode(**command_dict)
         async with self.communication_lock:
             self.log.debug(f"Sending command {st}")
-            assert self.writer is not None
-            self.writer.write(st.encode() + b"\r\n")
-            await self.writer.drain()
-            assert self.reader is not None
-            read_bytes = await asyncio.wait_for(
-                self.reader.readuntil(b"\r\n"), timeout=_TIMEOUT
-            )
+            try:
+                assert self.writer is not None
+                self.writer.write(st.encode() + b"\r\n")
+                await self.writer.drain()
+            except (asyncio.exceptions.TimeoutError, AssertionError) as e:
```

Comment:
I'm curious: what triggers the timeout error? Does StreamWriter.drain raise this? Do you have any control over this? (Or is this just copied from the read branch and not actually needed)?

Reply:
I copied from the read branch and tried without catching the timeout error. The unit tests pass so it indeed is not needed. Thanks for spotting that!